internal/daemon/ucrednet.go

```diff
@@ -49,7 +49,9 @@ func ucrednetGet(remoteAddr string) (pid int32, uid uint32, socket string, err e
 		}
 		socket = subs[3]
 	}
-	if pid == ucrednetNoProcess || uid == ucrednetNobody {
+	// Socket connections between process namespaces, pid appears to always be 0,
+	// but uid is visible.
+	if uid == ucrednetNobody {
```

Comment:
Do you have more details on this?  Is the pid coming from the other side as zero, and if so is that expected in the environment we're using it?  The comment feels a bit vague ("appears to") for something we want a strict understanding as it might bite us in the future. Also, a pid zero here could mean both an explicit pid=0, or a wrong match above, so we'll want to make a distinction here, and treat each case accordingly after we understand it properly.

Reply:
In kernel src:
net/core/socket.c cred_to_ucred call to
kernel/pid.c pid_vnr call to
kernel/pid.c pid_nr_ns which returns 0 if the process is not visible from the current process namespace.

Since the connecting process is in a non-related process namespace, the daemon process can't see the connecting process and therefore unable to resolve the pid.

Reply:
I will fix this tomorrow for the invalid case vs explicit zero (the case we care about here)

Reply:
Thanks, yeah that would be great. We just need a slightly more clear and strict comment explaining that then, and have a proper difference between bad/unset, and good that includes zero in such cases.

For bad data -1 would be okay, assuming the overflow is outside of the data range for the particular variable type (IOW, any valid pid falls under the max for the data type).